xm-commons-logging/src/main/java/com/icthh/xm/commons/logging/util/LogObjectPrinter.java

```diff
@@ -332,4 +430,38 @@ private static String printTypeAware(Object value) {
         return value.toString();
     }
 
+    public enum Level {
+        TRACE, DEBUG, INFO, WARN, ERROR, OFF_LOG
+    }
+
+    /**
+     * Log a message at the target level according to the specified format
+     * and arguments.
+     * @param log org.slf4j.Logger for keep the original loggerName
+     * @param level logging level
+     * @param format format string
+     * @param argArray list of arguments
+     */
+    public static void setLevelAndPrint(Logger log, Level level, String format, Object... argArray) {
```

Comment:
suggest method renaming: `setLevelAndPrint` --> `logWithLevel`
because:
a) this method does not et any level fo future
b) it does not print the message to a string but rather send it to the log.

Reply:
renamed